lms/djangoapps/teams/views.py

```diff
@@ -467,7 +471,19 @@ def get(self, request):
             'open_slots': ('team_size', '-last_activity_at'),
             'last_activity_at': ('-last_activity_at', 'team_size'),
         }
-        queryset = CourseTeam.objects.filter(**result_filter)
+
+        if not has_access(request.user, 'staff', course_key):
+            # hide private_managed courses from non-admin users that aren't members of those teams
+            private_topic_ids = [ts.teamset_id for ts in course_module.teamsets if
+                                 ts.teamset_type == TeamsetType.private_managed]
+            non_private_managed_memberships = CourseTeam.objects.filter(**result_filter).exclude(
+                topic_id__in=private_topic_ids)
+            private_managed_with_user = CourseTeam.objects.filter(topic_id__in=private_topic_ids,
+                                                                  membership__user__username=request.user)
+            queryset = non_private_managed_memberships | private_managed_with_user
+        else:
+            queryset = CourseTeam.objects.filter(**result_filter)
```

Comment:
Nit: consolidated suggestion on cleaning up the formatting and names just a little bit -
```python
if not has_access(request.user, 'staff', course_key):
    # hide private_managed courses from non-admin users that aren't members of those teams
    private_topic_ids = [
        ts.teamset_id for ts in course_module.teamsets if
        ts.teamset_type == TeamsetType.private_managed
    ]
    public_teams = CourseTeam.objects.filter(**result_filter).exclude(
        topic_id__in=private_topic_ids
    )
    private_managed_teams_of_user = CourseTeam.objects.filter(
        topic_id__in=private_topic_ids,
        membership__user__username=request.user
    )
    queryset = public_teams | private_managed_teams_of_user
else:
    queryset = CourseTeam.objects.filter(**result_filter)
```